test/libyul/yulOptimizerTests/fullSuite/reuse_vars_bug_in_simplifier.yul

```diff
@@ -0,0 +1,16 @@
+{
+	// This tests that a bug is fixed that was related to just taking
+	// values of SSA variables in account and not clearing them when
+	// a dependency was reassigned.
+	pop(foo())
+	function foo() -> x_9
+	{
+		x_9 := sub(1,sub(x_9,1))
+		mstore(sub(1,div(sub(x_9,1),sub(1,sub(x_9,1)))), 1)
+	}
+}
+// ----
+// fullSuite
+// {
+//     mstore(1, 1)
```

Comment:
Maybe stupid question, but why is `div(sub(x_9,1),sub(1,sub(x_9,1)))` 0?

Reply:
Ok, let's see: `x_9 = 0` at the beginning, so the first assignment to `x_9` results in `1 - (0-1) = 2`.
Then we have `sub(1,sub(x_9,1))`, i.e. `1 - (2-1) = 0`, and thus we divide by `0` which is defined to be `0` in the EVM.

Reply:
Ah, sorry, completely missed `x_9 = 0` and was wondering how `(1 - x_9) / x_9 = 0`